File: python_data-structures_algorithms/side_trail/fibonacci.py
def allFib(n):
    memo_1 = [0] * n
    for i in range(0, n):
        # print(str(i + 1) + ": " + str(fib(i, memo_1)))
        fib(i, memo_1)
        print("In Loop: " + str(memo_1))

def fib(n, memo):
    # print(str(memo))
    if n <= 0:
        memo[n] = 0
        return memo[n]
    elif n == 1:
        memo[n] = 1
        return memo[n]
    elif memo[n] > 0:
        return memo[n]
    digit_1 = fib(n - 1, memo)
    digit_2 = fib(n - 2, memo)
    memo[n] = digit_1 + digit_2
    



if __name__ == "__main__":
    # print("memo_1" in globals())
    allFib(10)

File: java_17/basics/Fibonacci.java
package java_17.basics;

import java.util.Arrays;

public class Fibonacci {
    
    public static void main(String[] args) {
        allFib(10);
    }
    
    public static void allFib(int n) {
        int[] memo = new int[n+1];
        for (int i = 0; i < n; i++) {
            System.out.println((i + 1) + ": " + fib(i, memo));
            System.out.println("In loop " + Arrays.toString(memo));
        }
        System.out.println(Arrays.toString(memo));
    }

    public static int fib(int n, int[] memo) {
        System.out.println(Arrays.toString(memo));
        if (n <= 0) {
            return 0;
        } else if (n == 1) {
            memo[n] = 1;
            return 1;
        } else if (memo[n] > 0) {
            return memo[n];
        }

        memo[n] = fib(n-1, memo) + fib(n - 2, memo);
        return memo[n];
    }
}
